cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -274,8 +280,14 @@ func generateRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o opti
 			generateDockerConfigJsonSecretConfigTo(regPullCredsAll, testCredentials, o.clusterName),
 		},
 	}
-	logrus.Infof("Creating new secret with 'to' of: %v", sc.To)
-	c.Secrets = append(c.Secrets, sc)
+	idx, _, _ := findSecretConfig(regPullCredsAll, o.clusterName, c.Secrets)
```

Comment:
Why is this not using `updateSecret()` ?

Reply:
`updateSecret`'s contract takes a `generatorFunction` that _only_ takes `options` as an argument. Here, we also need either the `secretbootstrap.Config` (or simply the list of `UserSecretsTargetClusters`) in order to properly generate the items.

Reply:
Could you not adapt it, like:

```go
func wrapped(c *secretbootstrap.Config) func(o options) error {
    return func(o options) error {
        for _, cluster := range c.UserSecretsTargetClusters {
            //...
        }
    }
}
```

Reply:
yeah that works!